### DISCUSSION TITLE
Trigger block of Table Row Creation for a Questionnaire

### DISCUSSION THREAD

**Discussion by Al-Blu** [Budibase/budibase#5064] [Help]
Hey, 
I am building a Rating App for a company where every User gets to rate a certain amount of Company Use Cases. I would like to create for each User, a block of Use-Case-Rows in a table and then they can rate their assigned Use Cases. 

How would I best approach triggering the creation of a block of rows assigned to the current User in BudiBase? Is there a template that has something similar?

My original approach was one UseCaseTable for all Users and a Results Table for all users, but I am struggling to show/hide UseCases that have been rated by the current user. I think this approach is too complicated.

Any guidance would be much appreciated.
Thanks

**Comment by mjashanks** [comment]
Hi @Al-Blu 

I think I know what you are trying to do here - there bit I'm unclear about is - how do the use cases get assigned to user? Does another user do it?

Let say another user creates the use-case. When they do so, an entry gets created in a `User Use Case` table.

Then, the user will go in an view their use cases - a list of items assigned to them in the `User Use Case` table.

Once they rate it and click `Submit`, then 
1. a new row gets created in the `Use case result` table,
2. The entry gets `deleted` from the `User use case` table.

**Reply by Al-Blu** [reply]
The Use Cases are already created by the Admin and exist in a table with the Admin as responsible. The UseCase would have fields for rating. Then every new User gets a copy or a block of rows with themselves as responsible. The User can only see and edit his own rows. That seems the easiest solution to me. And the Admin can export or show the results after everybody is finished rating.

**Reply by Al-Blu** [reply]
Also they way you mentioned with Use Case Result would work. That is what I kinda I have now. Nearly. I just need to register somehow that the User rated a particular Use Case and not show that one anymore, or mark it as 'Rated' with a color or similar....
I would need to query the Results Table

**When displaying the UseCase Table I would need a query something like this:
Get all UseCaseResults 
 where  {{ UseCaseResults.useCase ._id  }} EQUALS Binding {{ useCase._id }}
AND with those results
 {{ Current User.email }} Exists in results 
Hide component**
Not sure how I would do that query. Struggle to write it :-/
